test/e2e/utils.py

```
@@ -36,7 +48,12 @@ def wait_for_job_output(
         task_id=None,
     )
     output_paths_by_root = job_output_downloader.get_output_paths_by_root()
+    LOG.info(f"Output paths by root: {job_output_downloader.outputs_by_root}")
     # Download file and place it into the output_paths_by_root
+    if output_root_path is not None:
+        job_output_downloader.set_root_path(
+            list(output_paths_by_root.keys())[0], os.path.abspath(output_root_path)
+        )
```

What happens if there is more than one root?

---

Good question, it currently doesn't support more than one root. I've added a check to fail fast.